### PR TITLE
OCLOMRS-958:Errors from the backend should marked the field that caused the error

### PR DESCRIPTION
**Pull request by jwnasambu**
…ed the error

# JIRA TICKET NAME:
[Errors from the backend should marked the field that caused the error](https://issues.openmrs.org/browse/<OCLOMRS-958>)

# Summary: 
I modified the errorMsgReponse to returns an object rather than just a single string and added an useEffect() hook and update the form so that anything in the new error object field gets displayed at the bottom of the form.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
@jwamalwa  well done Julie, was this passing well locally?

**Comment by jwnasambu** [comment]
@HerbertYiga yes locally it builds.

**Comment by jwnasambu** [comment]
@ibacher Kind help please! Locally the changes build but when I make the PR the travis fails. Kindly how do I go about it?

File: src/redux/utils.ts [hunk 1]
```diff
@@ -103,6 +103,10 @@ export function invalidateCache(action: string, dispatch: Function) {
 }
 
 export function errorMsgResponse(response: any) {
+  let errorMsgResponse = { __detail__: "Action could not be completed. Please retry." };
+
+  if (response.data && has(response.data, "detail")) {
+    errorMsgResponse["__detail__"] = response.data.detail
```
Comment (ibacher) [inline src/redux/utils.ts:109]: The error in the build in Travis is caused by this line... essentially the `{` after the `if` is never closed. That said we don't need this `if` here at all. What you have on line 109 should be in place of what's on 117 and line 120-124 should look something like:

```ts
errorMsgResponse[key] =
  Array.isArray(response.data[key])
    ? response.data[key].join(",")
    : response.data[key];
```

**Pull request closed without merging**